awx/main/tasks.py

```diff
@@ -1473,14 +1474,22 @@ def build_env(self, job, private_data_dir, isolated=False, private_data_files=No
             if authorize:
                 env['ANSIBLE_NET_AUTH_PASS'] = network_cred.get_input('authorize_password', default='')
 
-        for env_key, folder, default in (
-                ('ANSIBLE_COLLECTIONS_PATHS', 'requirements_collections', '~/.ansible/collections:/usr/share/ansible/collections'),
-                ('ANSIBLE_ROLES_PATH', 'requirements_roles', '~/.ansible/roles:/usr/share/ansible/roles:/etc/ansible/roles')):
+        path_vars = (
+            ('ANSIBLE_COLLECTIONS_PATHS', 'collections_path', 'requirements_collections', '~/.ansible/collections:/usr/share/ansible/collections'),
+            ('ANSIBLE_ROLES_PATH', 'roles_path', 'requirements_roles', '~/.ansible/roles:/usr/share/ansible/roles:/etc/ansible/roles'))
+
+        config_values = read_ansible_config(job.project.get_project_path(), list(map(lambda x: x[1], path_vars)))
+
+        for env_key, config_setting, folder, default in path_vars:
             paths = default.split(':')
             if env_key in env:
                 for path in env[env_key].split(':'):
                     if path not in paths:
                         paths = [env[env_key]] + paths
+            elif config_setting in config_values:
+                for path in config_values[config_setting].split(':'):
+                    if path not in paths:
+                        paths = [config_values[config_setting]] + paths
```

Comment:
I'm thinking this might reverse the order of what was in `config_values[config_setting]`, will need to dig into that, if it is doing that, fix will be simple

Reply:
Could you do a simple test using `role_paths=path1,path2` (either in the global location or project location) and see that the resultant `job_env` has the right ordering? Have you confirmed this manually?

Reply:
oh wait, I see it now, you're not adding `path`, but adding `config_values[config_setting]` without the split.

This still isn't really airtight logic, I'd expect it to add duplicates in that case. Because you can't reliably check `if path not in paths` because you add entries to `paths` where the single entry contains multiple paths. While this will initially _work_, it will duplicate the entries.